diff --git a/org.spearce.jgit.test/tst/org/spearce/jgit/lib/RepositoryConfigTest.java b/org.spearce.jgit.test/tst/org/spearce/jgit/lib/RepositoryConfigTest.java
index da7e704..34ce04a 100644
--- a/org.spearce.jgit.test/tst/org/spearce/jgit/lib/RepositoryConfigTest.java
+++ b/org.spearce.jgit.test/tst/org/spearce/jgit/lib/RepositoryConfigTest.java
@@ -109,4 +109,11 @@ assertTrue(Arrays.equals(values.toArray(), repositoryConfig
        .getStringList("my", null, "somename")));
    checkFile(cfgFile, "[my]\n\tsomename = value1\n\tsomename = value2\n");
  }
+
+ public void test006_readCaseInsensitive() throws IOException {
+   final File path = writeTrashFile("config_001", "[Foo]\nBar\n");
+   RepositoryConfig repositoryConfig = new RepositoryConfig(null, path);
+BAD LINE
+   assertEquals(true, repositoryConfig.getBoolean("foo", null, "bar", false));
+   assertEquals("", repositoryConfig.getString("foo", null, "bar"));
+ }
 }
